takwimu/views.py

```diff
@@ -79,6 +97,53 @@ def get_context_data(self, **kwargs):
         context['sdgs'] = data
         return context
 
+    def get(self, request, *args, **kwargs):
+        sdg = request.GET.get('sdg', '')
+        country = request.GET.get('country', '')
+
+
+        self.create_indicator_sdg_map()
+
+        if sdg and country:
+            link = self.get_indicator_link(sdg, country)
+            if not link:
+                # return to sdg page and add a flag to show a pop up that no
+                # indicators matching that SDG were found
+                context = self.get_context_data()
+                context['sdg_found'] = False
+                return render(request, self.template_name, context)
+            else:
+                # follow link to profiles page
+                return HttpResponseRedirect(link)
+        return super(SDGIndicatorView, self).get(request, *args, **kwargs)
+
+    def get_indicator_link(self, sdg, country):
+        try:
+            indicator = self.indicator_sdg_map[sdg][country]
+            link = indicator['url'] + '#' +  indicator['topic_title']
+            return link
+        except KeyError:
+            return None
+
+    def create_indicator_sdg_map(self):
+        for page in ProfileSectionPage.objects.live():
+            country, _, _ = get_page_details(page)
+            country = slugify(country)
+            url = page.url
+            try:
+                for topic in page.body.stream_data:
+
+                    topic_title = slugify(topic['value']['title'])
+                    for indicator in topic['value']['indicators']:
+                        indicator_id = indicator['id']
+                        for widget in indicator['value']['widgets']:
+                            widget_id = widget['id']
+                            sdg = widget['value']['sdg']
```

Comment:
Just to confirm, aren't we tagging `Indicator`s? The implementation looks like we're tagging `Widget`s. If it's possible for two widgets in the _same_ indicator to belong to **two** different SDGs then cool. Otherwise, we should be tagging and redirecting to `Indicator`s